File: examples/key/load_local_key/load_local_key.py
from bloock.entity.key.key_type import KeyType
from bloock.client.key import KeyClient

if __name__ == "__main__":
    key_client = KeyClient()

    # Load a EcP256k key
    local_key = key_client.load_local_key(KeyType.EcP256k, "private key")

    # Load a Rsa2048 key
    local_key = key_client.load_local_key(KeyType.Rsa2048, "private key")

    # Load a Rsa3072 key
    local_key = key_client.load_local_key(KeyType.Rsa3072, "private key")

    # Load a Rsa4096 key
    local_key = key_client.load_local_key(KeyType.Rsa4096, "private key")

    # Load a Aes128 key
    local_key = key_client.load_local_key(KeyType.Aes128, "password")

    # Load a Aes256 key
    local_key = key_client.load_local_key(KeyType.Aes256, "password")

File: examples/key/load_local_key/load_local_key.go
package main

import (
	"github.com/bloock/bloock-sdk-go/v2/client"
	"github.com/bloock/bloock-sdk-go/v2/entity/key"
)

func main() {
	keyClient := client.NewKeyClient()

	// Load a EcP256k key
	_, _ = keyClient.LoadLocalKey(key.EcP256k, "private key")

	// Load a Rsa2048 key
	_, _ = keyClient.LoadLocalKey(key.Rsa2048, "private key")

	// Load a Rsa3072 key
	_, _ = keyClient.LoadLocalKey(key.Rsa3072, "private key")

	// Load a Rsa4096 key
	_, _ = keyClient.LoadLocalKey(key.Rsa4096, "private key")

	// Load a Aes128 key
	_, _ = keyClient.LoadLocalKey(key.Aes128, "password")

	// Load a Aes256 key
	_, _ = keyClient.LoadLocalKey(key.Aes256, "password")

	// Load a BJJ key
	_, _ = keyClient.LoadLocalKey(key.Bjj, "private key")
}
